udata/core/reuse/models.py

```diff
@@ -23,32 +27,99 @@ def hidden(self):
         return self(db.Q(private=True) | db.Q(datasets__0__exists=False) | db.Q(deleted__ne=None))
 
 
+def check_url_does_not_exists(url):
+    """Ensure a reuse URL is not yet registered"""
+    if url and Reuse.url_exists(url):
+        raise FieldValidationError(_("This URL is already registered"), field="url")
+
+
+@generate_fields(
+    searchable=True,
+    additionalSorts=[
+        {"key": "datasets", "value": "metrics.datasets"},
+        {"key": "followers", "value": "metrics.followers"},
+        {"key": "views", "value": "metrics.views"},
+    ],
+)
 class Reuse(db.Datetimed, WithMetrics, BadgeMixin, Owned, db.Document):
-    title = db.StringField(required=True)
-    slug = db.SlugField(
-        max_length=255, required=True, populate_from="title", update=True, follow=True
+    title = field(
+        db.StringField(required=True),
+        sortable=True,
+        show_as_ref=True,
+    )
+    slug = field(
+        db.SlugField(
+            max_length=255, required=True, populate_from="title", update=True, follow=True
+        ),
+        readonly=True,
+    )
+    description = field(
+        db.StringField(required=True),
+        markdown=True,
+    )
+    type = field(
+        db.StringField(required=True, choices=list(REUSE_TYPES)),
+        filterable={},
+    )
+    url = field(
+        db.StringField(required=True),
+        description="The remote URL (website)",
+        check=check_url_does_not_exists,
     )
-    description = db.StringField(required=True)
-    type = db.StringField(required=True, choices=list(REUSE_TYPES))
-    url = db.StringField(required=True)
     urlhash = db.StringField(required=True, unique=True)
     image_url = db.StringField()
-    image = db.ImageField(
-        fs=images, basename=default_image_basename, max_size=IMAGE_MAX_SIZE, thumbnails=IMAGE_SIZES
+    image = field(
```

Comment:
I think the image gets wrongly updated when updating the datasets list in the admin?